core_1-14/src/main/scala/org/scalacheck/ops/GenConfig.scala

```diff
@@ -0,0 +1,40 @@
+package org.scalacheck.ops
+
+import org.scalacheck.Gen
+import org.scalacheck.rng.Seed
+
+final class GenConfig private (
+  val seed: Seed,
+  val params: Gen.Parameters,
+  val retries: Int
+) {
+
+  private[this] def copy(
+    seed: Seed = this.seed,
+    params: Gen.Parameters = this.params,
+    retries: Int = this.retries
+  ): GenConfig = {
+    new GenConfig(seed, params, retries)
+  }
+
+  def withSeed(seed: Seed): GenConfig = copy(seed = seed)
+  def withParams(params: Gen.Parameters): GenConfig = copy(params = params)
+  def withRetries(retries: Int): GenConfig = copy(retries = retries)
+}
+
+object GenConfig {
+
+  def apply(seed: Seed, maxSize: Int, retries: Int): GenConfig = {
+    new GenConfig(seed, Gen.Parameters.default.withSize(maxSize), retries)
+  }
+
+  def apply(seed: Seed, maxSize: Int): GenConfig = {
+    GenConfig(seed, maxSize, 100)
+  }
+
+  def apply(seed: Seed): GenConfig = {
+    GenConfig(seed, 6)
+  }
+
+  val defaults: GenConfig = GenConfig(Seed(0))
+}
```

Comment:
This config object is new and designed to be forwards-compatible in a similar manner to contraband classes.